### DISCUSSION TITLE
Version v0.1.2

### DISCUSSION THREAD

**Discussion by ghost** [fortran-lang/fpm#282] [General]
Hi.

fpm-v0.1.2-windows-x86_64 works as expected.
![fpm-v0 1 2-windows-x86_64](https://user-images.githubusercontent.com/5913872/101559376-c1446380-399f-11eb-867c-a1ee79b250a4.png)

fpm-bootstrap-v0.1.2-windows-x86_64 doesn't create subdirectories.
![fpm-bootstrap-v0 1 2-windows-x86_64](https://user-images.githubusercontent.com/5913872/101559373-c0133680-399f-11eb-890b-4f3ccad37b5f.png)

Thanks

**Comment by urbanjost** [comment]
The two versions differ in this respect.  The Fortran version defaults to generating all the default directories as per the output from
```bash
fpm help new
```
or all the help text at once with
```bash
fpm help manual
```

But the original version had in mind that the most common usage case envisioned was for building modules and libraries so it defaults to
creating just /src.  You have to add `--with-executable` to create the app/ directory.  First impressions are always valuable, so let us know what behavior is preferred (and "none of the above" is a great answer too).

**Reply by ghost** [reply]
'fpm-bootstrap-v0.1.2-windows-x86_64 new project' creates the project's top level directory with 3 files inside: .gitignore, fpm.toml and README.md. The user has to create the src directory manually.

![fpm-bootstrap-v0 1 2-windows-x86_64_new](https://user-images.githubusercontent.com/5913872/101812291-efd65180-3af9-11eb-865f-79e39df0cd37.png)

![fpm-bootstrap-v0 1 2-windows-x86_64_src](https://user-images.githubusercontent.com/5913872/101812687-725f1100-3afa-11eb-9320-19721e9248e4.png)

I believe the fpm-fortran version is better for beginners, because it creates everything by default.

**Comment by urbanjost** [comment]
To help guide any feedback on this topic, consideration is being given to adding two more directories --  examples/ and benchmarks/ or something close to that.  So if `new` continues to default to everything it will create src/, app/, test/, example/, benchmarks/ and build/ when something is compiled. But a new project might only use src/ and build/, for example.  So a question arises whether the new directories should also be generated by default for consistency and simplicity or if that becomes too cluttered for those not planning on using those directories.

**Reply by ghost** [reply]
Yes, I believe the new directories should be generated by default for consistency. It doesn't cause any harm and it works by default.

**Reply by urbanjost** [reply]
Thanks!  I am thinking a poll for this would be worthwhile, but I might need someone else to activate it. The poll option is not coming up in my browser!

**Reply by milancurcic** [reply]
I don't know if GitHub Discussions have a poll option, but Discourse does. We did polls there in the past.

**Comment by urbanjost** [comment]
Some new users seem to be using both versions; and the differences available give an opportunity to compare different methods but also seem to be a point of confusion. I now think that as the Fortran version is developing rapidly and will hopefully address the issues with custom compiler and load options (that are my primary reasons for switching back to the bootstrap version myself) that the differences do not have to be resolved; but since the Fortran version can easily boot itself now ( I use a single-file version to start it up on a machine with nothing on it but git and gfortran with no problem) that it is time for them to be available independently (or nearly is) so I personally have changed my mind about trying to resolve the differences at this point.

Might be missing some opportunities to get good feedback from those trying both but there are only so many hours in the day.

**Comment by ghost** [comment]
Hello. I was able to reuse code with fpm, although in my particular use case I think the copy-paste approach is better.

```
[dependencies]
ode = { git = "https://github.com/brocolis/ode.git" }

```

**Comment by urbanjost** [comment]
Thanks! Even with a short exercise there are advantages to fpm. In less than a minute I could pull your work and run your test with three compilers. 
```text
 git clone "https://github.com/brocolis/ode.git"
fpm test --compiler ifort;fpm test --compiler nvfortran;fpm test --compiler ifort
 + build/ifort_debug/test/runTests 
   1.094281    
   1.207102    
   1.338356    
   1.487782    
   1.654989    
 + build/nvfortran_debug/test/runTests 
    1.094281    
    1.207102    
    1.338356    
    1.487782    
    1.654989    
 + build/ifort_debug/test/runTests 
   1.094281    
   1.207102    
   1.338356    
   1.487782    
   1.654989    
```
That might not be an advantage to you yet; but that made a relatively complex process very simple for someone you might be collaborating with or that you want to share your code with.  As the number of packages grows there will be other benefits. What if you wanted to add command line arguments, pull your data from an SQLite file, ...  So the advantages might seem small at this point (not having to directly work with compiler options or make(1) or Cmake(1) or ... ) but they can quickly grow substantial. Thanks so much for trying `fpm`!  Your feedback
is invaluable.

**Reply by ghost** [reply]
Yes. The process is a bit confused at this point, because I'm not proficient with Fortran/fpm/fpm-bootstrap/etc yet. I see fpm will soon become a key tool in the Fortran ecosystem. Thank you for developing fpm.

**Comment by urbanjost** [comment]
Well, I'm at the tail end of a long line of people contributing to `fpm` including early adoptors like yourself, so let's thank them all! I agree that  great things can come from `fpm`.
